test/e2e-go/e2e_go_olm_test.go

```diff
@@ -0,0 +1,136 @@
+		const operatorVersion = "0.0.1"
+
+		const (
+			OLMBundleValidationTest   = "olm-bundle-validation"
+			OLMCRDsHaveValidationTest = "olm-crds-have-validation"
+			OLMCRDsHaveResourcesTest  = "olm-crds-have-resources"
+			OLMSpecDescriptorsTest    = "olm-spec-descriptors"
+			OLMStatusDescriptorsTest  = "olm-status-descriptors"
+		)
+
+		It("should generate and run a valid OLM bundle and packagemanifests", func() {
+			By("generating the operator bundle")
+			// Turn off interactive prompts for all generation tasks.
+			replace := "operator-sdk generate kustomize manifests"
+			testutils.ReplaceInFile(filepath.Join(tc.Dir, "Makefile"), replace, replace+" --interactive=false")
+			err := tc.Make("bundle", "IMG="+tc.ImageName)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("building the operator bundle image")
+			// Use the existing image tag but with a "-bundle" suffix.
+			imageSplit := strings.SplitN(tc.ImageName, ":", 2)
+			bundleImage := imageSplit[0] + "-bundle"
+			if len(imageSplit) == 2 {
+				bundleImage += ":" + imageSplit[1]
+			}
+			err = tc.Make("bundle-build", "BUNDLE_IMG="+bundleImage)
+			Expect(err).NotTo(HaveOccurred())
+
+			if isRunningOnKind() {
+				By("loading the bundle image into Kind cluster")
+				err = tc.LoadImageToKindClusterWithName(bundleImage)
+				Expect(err).Should(Succeed())
+			}
+
+			By("adding the 'packagemanifests' rule to the Makefile")
+			err = tc.AddPackagemanifestsTarget()
+			Expect(err).Should(Succeed())
+
+			By("generating the operator package manifests")
+			err = tc.Make("packagemanifests", "IMG="+tc.ImageName)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("running the package manifests-formatted operator")
+			Expect(err).NotTo(HaveOccurred())
+			runPkgManCmd := exec.Command(tc.BinaryName, "run", "packagemanifests",
+				"--install-mode", "AllNamespaces",
+				"--version", operatorVersion,
+				"--timeout", "4m")
+			_, err = tc.Run(runPkgManCmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("running basic scorecard tests")
+			var scorecardOutput v1alpha3.TestList
+			runScorecardCmd := exec.Command(tc.BinaryName, "scorecard", "bundle",
+				"--selector=suite=basic",
+				"--output=json",
+				"--wait-time=40s")
+			scorecardOutputBytes, err := tc.Run(runScorecardCmd)
+			Expect(err).NotTo(HaveOccurred())
+			err = json.Unmarshal(scorecardOutputBytes, &scorecardOutput)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(scorecardOutput.Items)).To(Equal(1))
+			Expect(scorecardOutput.Items[0].Status.Results[0].State).To(Equal(v1alpha3.PassState))
+
+			By("running custom scorecard tests")
+			runScorecardCmd = exec.Command(tc.BinaryName, "scorecard", "bundle",
+				"--selector=suite=custom",
+				"--output=json",
+				"--wait-time=40s")
+			scorecardOutputBytes, err = tc.Run(runScorecardCmd)
+			Expect(err).NotTo(HaveOccurred())
+			err = json.Unmarshal(scorecardOutputBytes, &scorecardOutput)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(scorecardOutput.Items)).To(Equal(2))
```

Comment:
We talked about not running these tests as part of the Go e2e. Can we remove from here and do a follow-up to add them to the `test/integration` tests?

Reply:
I think we need to discuss that before to do a follow. These tests were added via tasks planned in the sprint which means that was discussed before to get done. Also, I think we should keep them. See that when we split the repos will make sense we have only the e2e tests to ensure the SDK features ( scorecard and olm ) on this repo. 

Also, see that the `test/integration` is not following the standards define so far and with them, we will NOT test it end to end and then, we will not able to ensure that the commands work with the project layout and with the makefile targets suggested in the docs.

Reply:
See; https://github.com/operator-framework/enhancements/pull/44/files#diff-0af5b52afc5f9db2b08e97937d991b40R233

Reply:
Ah I see now that this is just copied over from the existing file. Seems reasonable to keep for now, but we should add a TODO comment to split the scorecard custom test testing out.

We should also discuss if there's really a need for the custom test example at all. After all, the basic and olm tests are probably reasonable examples on their own.

Reply:
I will push soon the PR to split the scorecard.